tests/ast/nodes/test_evaluate_binop_int.py

```diff
@@ -0,0 +1,135 @@
+)
+import pytest
+
+from vyper import (
+    ast as vy_ast,
+)
+from vyper.exceptions import (
+    ZeroDivisionException,
+)
+
+
+@pytest.mark.fuzzing
+@settings(deadline=500)
+@given(
+    left=st.integers(min_value=-(2 ** 32), max_value=2 ** 32),
+    right=st.integers(min_value=-(2 ** 32), max_value=2 ** 32),
+)
+@pytest.mark.parametrize("op", "+-*/%")
+def test_binop_int128(get_contract, assert_tx_failed, op, left, right):
+    source = f"""
+@public
+def foo(a: int128, b: int128) -> int128:
+    return a {op} b
+    """
+    contract = get_contract(source)
+
+    vyper_ast = vy_ast.parse_to_ast(f"{left} {op} {right}")
+    old_node = vyper_ast.body[0].value
+    try:
+        new_node = old_node.evaluate()
+        is_valid = True
+    except ZeroDivisionException:
+        is_valid = False
+
+    if is_valid:
+        assert contract.foo(left, right) == new_node.value
+    else:
+        assert_tx_failed(lambda: contract.foo(left, right))
+
+
+@pytest.mark.fuzzing
+@settings(deadline=500)
+@given(
+    left=st.integers(min_value=0, max_value=2 ** 64),
+    right=st.integers(min_value=0, max_value=2 ** 64),
+)
+@pytest.mark.parametrize("op", "+-*/%")
+def test_binop_uint256(get_contract, assert_tx_failed, op, left, right):
+    source = f"""
+@public
+def foo(a: uint256, b: uint256) -> uint256:
+    return a {op} b
+    """
+    contract = get_contract(source)
+
+    vyper_ast = vy_ast.parse_to_ast(f"{left} {op} {right}")
+    old_node = vyper_ast.body[0].value
+    try:
+        new_node = old_node.evaluate()
+        is_valid = new_node.value >= 0
+    except ZeroDivisionException:
+        is_valid = False
+
+    if is_valid:
+        assert contract.foo(left, right) == new_node.value
+    else:
+        assert_tx_failed(lambda: contract.foo(left, right))
+
+
+@pytest.mark.fuzzing
+@settings(deadline=500)
+@given(
+    left=st.integers(min_value=2, max_value=245),
+    right=st.integers(min_value=0, max_value=16),
+)
+@example(left=0, right=0)
+@example(left=0, right=1)
+def test_binop_int_pow(get_contract, left, right):
+    source = """
+@public
+def foo(a: uint256, b: uint256) -> uint256:
+    return a ** b
+    """
+    contract = get_contract(source)
+
+    vyper_ast = vy_ast.parse_to_ast(f"{left} ** {right}")
+    old_node = vyper_ast.body[0].value
+    new_node = old_node.evaluate()
+
+    assert contract.foo(left, right) == new_node.value
+
+
+@pytest.mark.fuzzing
+@settings(deadline=500)
+@given(
+    values=st.lists(
+        st.integers(min_value=-256, max_value=256), min_size=2, max_size=10
+    ),
+    ops=st.lists(st.sampled_from("+-*/%"), min_size=11, max_size=11),
+)
+def test_binop_nested(get_contract, assert_tx_failed, values, ops):
+
+    variables = "abcdefghij"
+
+    return_value = " ".join(f"{a} {b}" for a, b in zip(variables[: len(values)], ops))
+    return_value = return_value.rsplit(maxsplit=1)[0]
+
+    source = f"""
+@public
+def foo({','.join(f'{i}: int128' for i in variables[:len(values)])}) -> int128:
```

Comment:
Do the `join` outside of the source string

Reply:
Oops! Thought I changed this in all of them, guess not.